swap/src/protocol/alice/event_loop.rs

```diff
@@ -257,7 +265,14 @@ impl EventLoop {
         .build()
         .await?;
 
-        tokio::spawn(async move { alice::run(swap).await });
+        let (remote, remote_handle) = alice::run(swap).remote_handle();
+        tokio::spawn(async move { remote.await });
```

Comment:
That should also do the trick, no?

```suggestion
        tokio::spawn(remote);
```